dataflow/pkg/db/drivers/mongo/mongo.go

```diff
@@ -700,7 +700,7 @@ func (ad *adapter) CreateJob(ctx context.Context, job *Job) (*Job, error) {
 		break
 	}
 	if i == 3 {
```

Comment:
what is the rationale behind trying 3 times ? i dont get it

Reply:
We created a random id for the job here. And there may be a conflict with those jobs which already exist in the database and if that happens we need to recreate it again.

Reply:
@sfzeng i am not sure trying 3 times will guarantee a unique ID. Given what you said, i can suggest 2 things, feel free to add on or minus

1. we use a sequence in the database, guaranteed to be unique
2. before insert, we check if ID already exists, use a different ID

Reply:
@rhsakarpos Here are 3 points:
1. In mongo, each document has an _id, as default, it's an ObjectId which is a 12 bytes binary BSON type. So I think it's a traditional way to use id like this, see [here](https://docs.mongodb.com/manual/reference/method/ObjectId/) please.
2. Mongo does not have auto-increment sequence.
3. If it failed 3 times, then we will return error to the end-user who can try again. And actually, the id will conflict with an extremely low probability, please check bson.NewObjectId().

Reply:
BTW, checking if ID already exists and then use a different ID, it's actually two steps, which may bring concurrency problem, whiling inserting directly and check if ID already exists by error is one step will not.